src/test/java/software/aws/neptune/gremlin/mock/MockGremlinDatabase.java

```diff
@@ -16,68 +16,87 @@
 
 package software.aws.neptune.gremlin.mock;
 
-import org.apache.commons.lang3.SystemUtils;
-import java.io.BufferedReader;
+import org.apache.tinkerpop.gremlin.jsr223.ScriptFileGremlinPlugin;
+import org.apache.tinkerpop.gremlin.server.GremlinServer;
+import org.apache.tinkerpop.gremlin.server.Settings;
+import org.apache.tinkerpop.gremlin.structure.io.Storage;
+
 import java.io.File;
-import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public class MockGremlinDatabase {
-    private static final String WINDOWS_SERVER = "gremlin-server.bat";
-    private static final String NIX_SERVER = "gremlin-server.sh";
-    private static final String SERVER_PATH =
-            "./gremlin-server/target/apache-tinkerpop-gremlin-server-3.5.0-SNAPSHOT-standalone/bin/";
-    private static final String SERVER_COMMAND = SERVER_PATH + NIX_SERVER;
-    private static final String START_COMMAND = String.format("%s start", SERVER_COMMAND);
-    private static final String STOP_COMMAND = String.format("%s stop", SERVER_COMMAND);
-    private static Process serverProcess = null;
 
+    private static GremlinServer server;
 
     /**
-     * Simple function to start the database.
-     *
-     * @throws IOException          thrown if command fails.
-     * @throws InterruptedException thrown if command fails.
+     * Starts a new instance of Gremlin Server.
      */
-    public static void startGraph() throws IOException, InterruptedException {
-        final String output = runCommand(START_COMMAND);
-        if (output.startsWith("Server already running with PID")) {
-            return;
-        }
-        Thread.sleep(10000);
+    public static void startServer() throws Exception {
+        final InputStream stream = MockGremlinDatabase.class.getResourceAsStream("/gremlin-server.yaml");
+        final Settings settings = Settings.read(stream);
+        rewritePathsInGremlinServerSettings(settings);
+        server = new GremlinServer(settings);
+        server.start().join();
     }
 
     /**
-     * Simple function to shut the database down.
-     *
-     * @throws IOException thrown if command fails.
+     * Stops a current instance of Gremlin Server.
      */
-    public static void stopGraph() throws IOException, InterruptedException {
-        runCommand(STOP_COMMAND);
-        Thread.sleep(10000);
+    public static void stopServer() {
+        server.stop().join();
+        server = null;
     }
 
-    private static String runCommand(final String command) throws IOException {
-        if (SystemUtils.IS_OS_WINDOWS) {
-            runWindowsCommand(command);
-        } else {
-            serverProcess = Runtime.getRuntime().exec(command);
+    /**
+     *  Tests from maven will pass this value in via failsafe plugin. basically we want to home this in on the
```

Comment:
Nit: This added method uses Maven-centric terminology when this is a Gradle project. You should probably change this.

Reply:
Ah yes, this was a helper method copied from TP test set up. I'll update.